packages/ui/src/components/layout-v2.tsx

```diff
@@ -0,0 +1,199 @@
+import { FC, HTMLAttributes, ReactNode } from 'react'
+
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+const gridVariants = cva('grid', {
+  variants: {
+    align: {
+      start: 'items-start',
+      center: 'items-center',
+      end: 'items-end',
+      baseline: 'items-baseline',
+      stretch: 'items-stretch'
+    },
+    justify: {
+      start: 'justify-start',
+      center: 'justify-center',
+      end: 'justify-end',
+      between: 'justify-between'
+    }
+  }
+})
+
+const flexVariants = cva('flex', {
+  variants: {
+    direction: {
+      row: 'flex-row',
+      column: 'flex-col',
+      'row-reverse': 'flex-row-reverse',
+      'column-reverse': 'flex-col-reverse'
+    },
+    align: {
+      start: 'items-start',
+      center: 'items-center',
+      end: 'items-end',
+      baseline: 'items-baseline',
+      stretch: 'items-stretch'
+    },
+    justify: {
+      start: 'justify-start',
+      center: 'justify-center',
+      end: 'justify-end',
+      between: 'justify-between'
+    },
+    wrap: {
+      nowrap: 'flex-nowrap',
+      wrap: 'flex-wrap',
+      'wrap-reverse': 'flex-wrap-reverse'
+    }
+  },
+  defaultVariants: {
+    direction: 'row'
+  }
+})
+
+const spacingVariants = cva('', {
+  variants: {
+    spacing: {
+      small: 'var(--cn-spacing-2)',
+      medium: 'var(--cn-spacing-4)',
+      large: 'var(--cn-spacing-6)'
+    }
+  },
+  defaultVariants: {
+    spacing: 'medium'
+  }
+})
+
+const gapVariants = cva('', {
+  variants: {
+    gap: {
+      none: 'gap-0',
+      xs: 'gap-1',
+      sm: 'gap-2',
+      md: 'gap-4',
+      lg: 'gap-6',
+      xl: 'gap-8'
+    },
+    gapX: {
+      none: 'gap-x-0',
+      xs: 'gap-x-1',
+      sm: 'gap-x-2',
+      md: 'gap-x-4',
+      lg: 'gap-x-6',
+      xl: 'gap-x-8'
+    },
+    gapY: {
+      none: 'gap-y-0',
+      xs: 'gap-y-1',
+      sm: 'gap-y-2',
+      md: 'gap-y-4',
+      lg: 'gap-y-6',
+      xl: 'gap-y-8'
+    }
+  }
+})
+
+type GapSize = 'none' | 'xs' | 'sm' | 'md' | 'lg' | 'xl'
+
+interface LayoutProps {
+  children?: ReactNode
+  className?: string
+  gap?: GapSize
+  gapX?: GapSize
+  gapY?: GapSize
+  as?: 'div' | 'span'
```

Comment:
why restrict this to just div and span? i could see semantic html tags being useful as well.. like `article`, `nav` etc. we can remove this type, and allow any html element.